worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala

```diff
@@ -821,18 +983,28 @@ class PushDataHandler(val workerSource: WorkerSource) extends BaseMessageHandler
       fileWriter.decrementPendingWrites()
       return
     }
-    val writePromise = Promise[Unit]()
+    val writePromise = Promise[Array[StatusCode]]()
     writeLocalData(Seq(fileWriter), body, shuffleKey, isPrimary, None, writePromise)
     // for primary, send data to replica
     if (location.hasPeer && isPrimary) {
       // to do
       Try(Await.result(writePromise.future, Duration.Inf)) match {
-        case Success(_) => wrappedCallback.onSuccess(ByteBuffer.wrap(Array[Byte]()))
+        case Success(result) =>
+          if (result.exists(_ != StatusCode.SUCCESS)) {
+            wrappedCallback.onFailure(new CelebornIOException("Write data failed!"))
```

Comment:
Use result(0) for Exception message?

Reply:
`result.length` is not always be 1. 
In addition, if `result(x) != StatusCode.SUCCESS`, it will be StatusCode.HARD_SPLIT, however,  MapPartitionPushData doesn't support HARD_SPLIT and HARD_SPLIT can not be the Exception message.

Reply:
if writePromise is success, result.length should be 1 for map partition as there is only one fileWriter